saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/algorithm/SavedSitesSyncPersisterAlgorithm.kt

```diff
@@ -208,44 +214,65 @@ class RealSavedSitesSyncPersisterAlgorithm @Inject constructor(
                 val favouriteEntry = entries.find { it.id == child }
                 if (favouriteEntry == null) {
                     Timber.d("Sync-Bookmarks: id $child not present in the payload, has it moved position?")
-                    val storedFavorite = repository.getFavoriteById(child)
+                    val storedFavorite = syncSavedSitesRepository.getFavoriteById(child, favoriteFolder)
                     if (storedFavorite == null) {
                         Timber.d("Sync-Bookmarks: id $child not present locally as Favourite")
-                        val storedBookmark = repository.getBookmarkById(child)
+                        val storedBookmark = savedSitesRepository.getBookmarkById(child)
                         if (storedBookmark == null) {
                             Timber.d("Sync-Bookmarks: id $child not present locally as Bookmark either, omitting")
                         } else {
                             Timber.d("Sync-Bookmarks: id $child is a Bookmark locally, adding it as Favourite")
-                            repository.insertFavorite(url = storedBookmark.url, title = storedBookmark.title)
-                            // repository.markBookmarkAsFavourite(id)
+                            syncSavedSitesRepository.insertFavorite(
+                                url = storedBookmark.url,
+                                title = storedBookmark.title,
+                                favoriteFolder = favoriteFolder,
+                            )
+                            // savedSitesRepository.markBookmarkAsFavourite(id)
                         }
                     } else {
                         if (storedFavorite.position != position) {
                             Timber.d("Sync-Bookmarks: id $child present locally and moved from position ${storedFavorite.position} to $position")
-                            processFavourite(conflictResolution, storedFavorite.copy(position = position, lastModified = lastModified))
+                            processFavourite(
+                                conflictResolution,
+                                storedFavorite.copy(position = position, lastModified = lastModified),
+                                favoriteFolder,
+                            )
                         } else {
                             Timber.d("Sync-Bookmarks: id $child present locally but in the same position")
                         }
                     }
                 } else {
                     val favourite = decryptFavourite(favouriteEntry, position, lastModified)
-                    processFavourite(conflictResolution, favourite)
+                    processFavourite(conflictResolution, favourite, favoriteFolder)
+                }
+            }
+            Timber.d("Sync-Bookmarks: comparing local favourites vs remote ones")
+            val storedFavourites = syncSavedSitesRepository.getFavoritesSync(favoriteFolder)
```

Comment:
nit: remove this. the extra call is not necessary if we are not dealing with favourites at the moment. Is there a task to take care of this in the future?